src/rgw/driver/rados/rgw_notify.cc

```diff
@@ -84,6 +84,26 @@ auto make_stack_allocator() {
 
 const std::string Q_LIST_OBJECT_NAME = "queues_list_object";
 
+struct PublishCommitCompleteArg {
+
+    PublishCommitCompleteArg(std::string _queue_name, const DoutPrefixProvider *_dpp)
+            : queue_name{std::move(_queue_name)}, dpp{_dpp} {}
+
+    std::string queue_name;
+    const DoutPrefixProvider *dpp;
+};
+
+void publish_commit_completion(rados_completion_t completion, void *arg) {
+    auto *comp_obj = static_cast<librados::AioCompletionImpl *>(completion);
+    auto *pcc_arg = static_cast<PublishCommitCompleteArg *>(arg);
```

Comment:
probably better to use an `std:unique_ptr` here.
also, can you dio static cast from `void*`? i think that reinterpret cast would work better here

Reply:
> I believe both can be used to convert from void* to T*

did not know that. thanks!